core/client/hdfs/src/main/java/alluxio/hadoop/AbstractFileSystem.java

```diff
@@ -458,16 +457,32 @@ public void initialize(URI uri, org.apache.hadoop.conf.Configuration conf) throw
     mStatistics = statistics;
     mUri = URI.create(mAlluxioHeader);
 
-    boolean masterAddIsSameAsDefault = checkMasterAddress();
-
-    if (sInitialized && masterAddIsSameAsDefault) {
+    // Create the master inquire client that we would have after merging the hadoop conf into
+    // Alluxio Configuration.
+    MasterInquireClient.Factory.Config inquireClientConf =
+        MasterInquireClient.Factory.Config.defaults();
+    inquireClientConf.setZookeeperEnabled(conf.getBoolean(PropertyKey.ZOOKEEPER_ENABLED.getName(),
+        inquireClientConf.isZookeeperEnabled()));
+    inquireClientConf.setZookeeperAddress(
+        conf.get(PropertyKey.ZOOKEEPER_ADDRESS.getName(), inquireClientConf.getZookeeperAddress()));
+    inquireClientConf.setElectionPath(conf.get(PropertyKey.ZOOKEEPER_ELECTION_PATH.getName(),
+        inquireClientConf.getElectionPath()));
+    inquireClientConf.setLeaderPath(
+        conf.get(PropertyKey.ZOOKEEPER_ELECTION_PATH.getName(), inquireClientConf.getLeaderPath()));
+    inquireClientConf.setConnectHost(mUri.getHost());
+    inquireClientConf.setConnectPort(mUri.getPort());
+    MasterInquireClient configClient = MasterInquireClient.Factory.create(inquireClientConf);
```

Comment:
- do we need to return this client after comparison?
- is this an overkill (creating a new client) just for equality check? any possible performance hit?

Reply:
- We do not need to return the client after comparison.
- Good call, I double checked and found that the constructor for the zookeeper inquire client was expensive. I've updated it to be light (just setting fields) and calling other basic constructors.